.github/workflows/build.yml

```diff
@@ -76,3 +76,33 @@ jobs:
           git add .
           git commit -s -m "chore: self mutation"
           git push origin HEAD:$PULL_REQUEST_REF
+  rosetta-compat:
+    runs-on: ubuntu-latest
+    permissions: {}
+    env:
+      CI: "true"
+      NODE_OPTIONS: --max_old_space_size=4096
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      - name: Install dependencies
+        run: yarn install --check-files
+      - name: Install Rosetta version
+        run: yarn add --dev jsii-rosetta@${{ matrix.rosetta }}
+      - name: Check Rosetta version
+        run: test $(npx jsii-rosetta --version) = "${{ matrix.rosetta }}"
+      - name: compile+test
+        run: |-
+          npx projen compile
+          npx projen test
```

Comment:
I suspect running `npx projen ...` will run `yarn install` which may cause the `jsii-rosetta` to be pruned or reverted... As a result, I'd actually rather have the `jsii-rosetta --version` test done **after** this as well...

Also - this doesn't quite test that everything works "as intended" when it was built with `jsii-rosetta@5.1.x` but is running with `jsii-rosetta@5.2.x` (hypothetically).

Reply:
projen doesn't run an install here, but I sure can move the check. 👍

Reply:
> Also - this doesn't quite test that everything works "as intended" when it was built with jsii-rosetta@5.1.x but is running with jsii-rosetta@5.2.x (hypothetically).

Yes ideally this would take the same approach as we e.g. do with `aws-cdk-lib`. Where we'd build the package to a tarball. Then put that tarball + desired rosetta version in an empty workspace and then run an integration test suite.
But the package isn't structured that way (yet) so that would be a lot more work. **Is it worth it?**
'Cause we we don't really build this package "with" `jsii-rosetta`. We run `tsc` and that's about it. Not bundling or whatsoever. Technically the current test suite doesn't even test the artifact, as it's running with `ts-node`.